### PR TITLE
Fix: disables Order Proposal plugin in versions below 2.0.2 to force update

### PR DESCRIPTION
**Pull request by alexmigf**
Regarding this future implementation: https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/705

### CODE REVIEW

**Comment by MohamadNateqi** [comment]
What do you think about always checking for the Proposal version? Not only in the upgrade, to avoid crashing stores?

**Comment by alexmigf** [comment]
I think the upgrade routine runs also when doing new installs.

**Comment by MohamadNateqi** [comment]
The only time it can crash the site is when they install an old version of the Order Proposal while they have the latest version of the PDF Invoice plugin. Right?
In this case, the only way to prevent activating that is to always check the Proposal version. Do you think it's a good idea to do this?

**Comment by alexmigf** [comment]
How do you propose to do that?

**Comment by MohamadNateqi** [comment]
We can add a function to the "plugins_loaded" hook to check dependent plugins. What do you think?

**Comment by alexmigf** [comment]
What that differs from including in the upgrade routine?

**Comment by MohamadNateqi** [comment]
If the customer doesn't have the Order Proposal plugin installed and then installs it when they have the latest version of PDF Invoices, the upgrade routine won't happen, and this could lead to the crash you mentioned [here](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/705#discussion_r1478394392). I think this is the only scenario that could lead to a crash.
So, I suggested directly checking the installed version of the Order Proposal to prevent this scenario.

**Comment by alexmigf** [comment]
If they install the Proposal plugin I think they will do it with the latest version, but doesn't hurt adding that as well.

**Comment by MohamadNateqi** [comment]
Would it solve the crash issue you mentioned? Or are there also other scenarios that could lead to crash stores?

**Comment by alexmigf** [comment]
The only scenarios I'm seeing now are:

* The Proposal plugin running below 2.0.2
* Other third party plugin that extends our document class not declaring he `init_number` correctly. We need to check compatible plugins about this, specially the [Italian addon](https://github.com/alexmigf/woocommerce-pdf-italian-add-on) and provide them with changes if necessary.
* Fix here also https://github.com/wpovernight/wcpdf-quotation

I think a few users might receive the fatal error, that's why I'm providing a custom message with a documentation link so they could fix on their end if that happens.

**Comment by MohamadNateqi** [comment]
So It seems like there is no way to avoid this situation!
Then maybe not so much necessary to add my suggestion [here](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/706#issuecomment-1927374823).

File: woocommerce-pdf-invoices-packingslips.php [hunk 1]
```diff
@@ -542,6 +543,34 @@ public function legacy_addon_notices() {
...
+		if ( get_transient( $transient_name ) ) {			
+			ob_start();
+			?>
+			<div class="notice notice-warning">
+				<p><?php _e( 'Attention! We detected that your WooCommerce Order Proposal plugin is running a version below 2.0.2. In the next version of the PDF Invoices & Packing Slips for WooCommerce plugin, we are making a drastic change that requires at least version 2.0.2 of the Proposals plugin. To avoid clashes, we\'ve temporarily deactivated it—feel free to update to version 2.0.2 or higher. Thank you!', 'woocommerce-pdf-invoices-packing-slips' ); ?></p>
+				<p><a href="<?php echo esc_url( wp_nonce_url( add_query_arg( array( $query_arg => true ) ), 'hide_order_proposal_below_2_0_2_notice_nonce' ) ); ?>"><?php _e( 'Hide this message', 'woocommerce-pdf-invoices-packing-slips' ); ?></a></p>
+			</div>
+			<?php
+			echo wp_kses_post( ob_get_clean() );
+		}
```
Comment (MohamadNateqi) [inline woocommerce-pdf-invoices-packingslips.php:560]: Should it check the Order Proposal plugin version before displaying this notice?

**Review by dwalkerpriv — APPROVED** [review]
Appears appropriately and does disable the plugin 👍 

Screenshot 2024-02-05 152734

**Comment by alexmigf** [comment]
I have changed my mind, I think we should play safe and keep the original function for now. Closing.

**Pull request closed without merging**